subprojects/core/src/main/java/org/gradle/cache/internal/GradleUserHomeCacheCleanupEnablement.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache.internal;
+
+import org.gradle.cache.CleanupAction;
+import org.gradle.initialization.GradleUserHomeDirProvider;
+import org.gradle.util.internal.GUtil;
+
+import java.io.File;
+import java.util.Properties;
+
+public class GradleUserHomeCacheCleanupEnablement implements CacheCleanupEnablement {
```

Comment:
I think (especially if you choose to explore the suggestion about removing `CacheCleanupEnablement`) that `Enablement` is a poor name for this type.  I don't mind noun-ifying the word enable, but think a `Factory` or `Decorator` suffix would better describe what it does: provides a factory method for decorating `CleanupAction`s with a progress monitor and checking if the property is enabled.

If it's primary purpose is to create the `CleanupAction` if required, wouldn't `GradleUserHomeCacheConditionalCleanupActionFactory`?  Or maybe `GradleUserHomeCacheMonitoredConditionalCleanupActionDecorator`?  Those are also mouthfuls, but I think a little clearer.

Reply:
Good point.  This evolved from just pulling out the "is it enabled" capability into something centralized, and only later took on the decorating functionality.  `Decorator` is a better descriptor of its function, so everything has been renamed correspondingly.  Thanks for the suggestion.